Code/GraphMol/SmilesParse/SmartsWrite.cpp

```diff
@@ -260,6 +260,7 @@ std::string getAtomSmartsSimple(const QueryAtom *qatom,
         res << "5";
         break;
     }
+    needParen = true;
```

Comment:
Is there a test for this?

Reply:
And actually the others now I looked for them.

Reply:
The tests I added at l126 of smatest.cpp, which output the SMARTS and then make sure it can be read back in again, exercise this.